### PR TITLE
[Fix](Planner) add date/month/year add function fe constant fold with dateV2/dateTimeV2

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
## Proposed changes

cherry-pick: #34483

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/rewrite/FEFunctions.java [hunk 1]
```diff
@@ -114,18 +124,54 @@ public static DateLiteral yearsAdd(LiteralExpr date, LiteralExpr year) throws An
...
     @FEFunction(name = "months_add", argTypes = { "DATETIME", "INT" }, returnType = "DATETIME")
     public static DateLiteral monthsAdd(LiteralExpr date, LiteralExpr month) throws AnalysisException {
         DateLiteral dateLiteral = (DateLiteral) date;
         return dateLiteral.plusMonths((int) month.getLongValue());
     }
 
+    @FEFunction(name = "months_add", argTypes = { "DATEV2", "INT" }, returnType = "DATEV2")
+    public static DateLiteral monthsAddDateV2(LiteralExpr date, LiteralExpr month) throws AnalysisException {
+        DateLiteral dateLiteral = (DateLiteral) date;
+        return dateLiteral.plusYears(month.getLongValue());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/rewrite/FEFunctions.java:148]:
```suggestion
        return dateLiteral.plusMonths(month.getLongValue());
```

**Review by morrySnow — CHANGES_REQUESTED** [review]

**Comment by morrySnow** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]
LGTM

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Comment by LiBinfeng-01** [comment]
run feut

**Pull request merged**